nexus/types/src/deployment/planning_input.rs

```diff
@@ -333,6 +333,32 @@ pub struct SledDetails {
     pub resources: SledResources,
 }
 
+impl SledDetails {
+    /// Returns `Some(reason)` if the sled needs its zones to be expunged,
+    /// based on the policy and state.
+    pub fn needs_zone_expungement(&self) -> Option<ZoneExpungeReason> {
+        match self.state {
+            SledState::Active => {}
+            SledState::Decommissioned => {
+                // A decommissioned sled that still has resources attached to
+                // it is an illegal state, but representable. If we see a sled
```

Comment:
This comment trails off.

Reply:
Haha that was because as I was writing this comment I started thinking about the situations where this could happen, and went down a long rabbit hole. I'll fix this.

Reply:
Done.